WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionWebViewClient.java

```diff
@@ -0,0 +1,137 @@
+package org.wordpress.android.ui;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import org.wordpress.android.analytics.AnalyticsTracker;
+import org.wordpress.android.fluxc.model.SiteModel;
+import org.wordpress.android.fluxc.store.AccountStore;
+import org.wordpress.android.fluxc.store.SiteStore;
+import org.wordpress.android.login.LoginMode;
+import org.wordpress.android.ui.accounts.LoginActivity;
+import org.wordpress.android.ui.prefs.AppPrefs;
+import org.wordpress.android.util.AppLog;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+
+class JetpackConnectionWebViewClient extends WebViewClient {
+
+    private static final String LOGIN_PATH = "/wp-login.php";
+    private static final String ADMIN_PATH = "/wp-admin/admin.php";
+    private static final String REDIRECT_PARAMETER = "redirect_to=";
+    private static final int REQUEST_CODE = 1;
```

Comment:
Although it's not necessary to change this, I wanted to point out [RequestCodes.java](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/RequestCodes.java) which contains most of the app's request codes.